managed-entitlements/src/managed_entitlements_checkout_borrow_sample.py

```diff
@@ -0,0 +1,147 @@
+    return lm_client.create_license(
+        LicenseName="My License",
+        ProductName="My Product",
+        Beneficiary="My Beneficiary",
+        ConsumptionConfiguration={
+            "ProvisionalConfiguration": {
+                "MaxTimeToLiveInMinutes": 60
+            },
+            "BorrowConfiguration": {
+                "MaxTimeToLiveInMinutes": 1440,
+                "AllowEarlyCheckIn": True
+            },
+        },
+        Entitlements=[{
+            "Name":"Users",
+            "Unit":"Count",
+            "MaxCount": 1000,
+            "Overage": False,
+            "AllowCheckIn": True
+        }],
+        HomeRegion=region,
+        Issuer={
+            "Name":"My Company",
+            "SignKey": SignKey
+        },
+        ProductSKU="3688750d24200a4daa2e046732580b30",
+        Validity={
+            "Begin": "2021-04-12T00:00:00Z"
+        },
+        LicenseMetadata=[{
+            "Name":"ProductName",
+            "Value":"My awesome product"
+        }],
+        ClientToken=str(uuid.uuid4())
+    )
+
+def checkout_borrow_license(LicenseArn):
+    """Borrow a license for offline usage
+
+    The API returns an SignedToken which is a JWT token encoded using algorithm
+    PS384. The data is essentially the same returned as response of this object.
+    For more details: https://docs.aws.amazon.com/license-manager/latest/APIReference/API_CheckoutBorrowLicense.html
+    """
+
+    lm_client = boto3.client('license-manager', region)
+    return lm_client.checkout_borrow_license(
+        LicenseArn=LicenseArn,
+        Entitlements=[{
+            "Name":"Users",
+            "Value": "1",
+            "Unit":"Count"
+        }],
+        NodeId="MyNodeId",
+        DigitalSignatureMethod="JWT_PS384",
+        ClientToken=str(uuid.uuid4()).replace("-","")
+    )
+
+
+def create_cmk(desc='Customer Master Key'):
+    """Create a KMS Customer Master Key
+
+    The created CMK is a Customer-managed key stored in AWS KMS.
+    """
+
+    kms_client = boto3.client('kms', region)
+    response = kms_client.create_key(
+                                    Description=desc,
+                                    CustomerMasterKeySpec="RSA_4096",
+                                    KeyUsage="SIGN_VERIFY"
+                                )
+
+    return response['KeyMetadata']['Arn']
+
+def get_public_key(KeyId):
+    """Get CMK PublicKey
+    """
+
+    kms_client = boto3.client('kms', region)
+    response = kms_client.get_public_key(KeyId=KeyId)
+
+    return response['PublicKey']
+
+def main(command_line=None):
+    print("Start of the sample model for checkout borrow license")
+    # Initially, you should create an asymmetric key that License Manager uses to
+    # sign borrow license data. The algorithms are based on JWT standard and all
+    # major languages has public libraries to verify signatures. For this sample,
+    # we are using PyJwt
+    SignKey=create_cmk()
```

Comment:
We need to be clear that this needs to be created in the region where you are doing checkoutborrow. Key and checkoutborrow should be in the same region

Reply:
Addressed in new revision